### ISSUE TITLE
GHRS: Graph-based Hybrid Recommendation System with Application to Movie Recommendation

### ISSUE THREAD

**Issue by hello-im-yj** [One-Day-One-Paper/One-Day-One-Paper#316]
This paper proposes a recommender system method using a graph-based model associated with the similarity of users’ ratings in combination with users’ demographic and location information. By utilizing the advantages of Autoencoder feature extraction, we extract new features based on all combined attributes. Using the new set of features for clustering users, our proposed approach (GHRS) outperformed many existing recommendation algorithms on recommendation accuracy. Also, the method achieved significant results in the cold-start problem. 

### 3.1. Architecture
<img width="798" alt="스크린샷 2022-10-30 오후 11 52 13" src="https://user-images.githubusercontent.com/63237947/198885255-1798fc22-1237-4847-99d1-8043432e60e5.png">

1. In the first step, we build a graph with the number of users’ as nodes. Two users will be connected based on their similarities. The edge connects a pair of users who have more than $alpha$ percent of items with similar ratings. 
2. In the second step, a set of information will be extracted from the similarity graph for each user. For instance, we compute PageRank of the nodes, degree centrality, closeness centrality, the shortest-path betweenness centrality, load centrality, and the average degree of each node’s neighborhood in the graph. As a result, this matrix relies on the different data processing magnitude using a preference-based collaborative approach. 
3. In the third step, we combine side information such as gender and age with graph-based features to retrieve the most relevant movies for users. Therefore, we have one combined matrix from different types of features, which is then used as the Autoencoder stage input. 
4. In the fourth step, we apply the Autoencoder to extract new features and reduce the dimension. It includes selecting a proper optimizer, using a proper loss function and neural network architecture, and preventing the overfitting issue. 
5. In the fifth step, we utilize the new features encoded by Autoencoder for user clustering, using the K-means algorithm to create a small number of peer groups. Itincludes finding an appropriate number of clusters for each dataset.
6. In the sixth step, we assign new users to clusters based on encoded features and compute the new item rating base on similarity with other items.
7. In the seventh step, we compute the estimated rates of all items for each user according to its cluster’s average rating.
